crates/cashu/src/nuts/nut10.rs

```diff
@@ -105,6 +108,393 @@ impl Secret {
     }
 }
 
+/// Get the relevant public keys and required signature count for P2PK or HTLC verification
+///
+/// Takes into account locktime - if locktime has passed, returns refund keys,
+/// otherwise returns primary pubkeys/hash path. Returns (preimage_needed, pubkeys, required_sigs).
+/// For P2PK, preimage_needed is always false. For HTLC, preimage_needed is true before locktime.
+pub fn get_pubkeys_and_required_sigs(
```

Comment:
Does this need to be public? Maybe can be just public to the crate seems like an internal fn.